README.md

```diff
@@ -21,65 +21,116 @@ This library implements SPDX parsers, convertors, validators and handlers in Pyt
+As always you should work in a virtualenv (venv). You can install a local clone
+of this repo with `yourenv/bin/pip install .` or install it from PyPI with
+`yourenv/bin/pip install spdx-tools`. Note that on Windows it would be `Scripts`
+instead of `bin`.
 
 # How to use
 
 ## Command-line usage:
 
 1. **PARSING/VALIDATING** (for parsing any format):
 
-* Use `pyspdxtools -i <filename>` where `<filename>` is the location of the file.              
-  If you are using a source distribution, try running: `pyspdxtools -i tests/data/formats/SPDXJSONExample-v2.3.spdx.json`.
+* Use `pyspdxtools -i <filename>` where `<filename>` is the location of the file. The input format is inferred automatically from the file ending.
 
-* Or you can use `pyspdxtools` only, and it will automatically prompt/ask for the `input file path`.
+* If you are using a source distribution, try running:  
+  `pyspdxtools -i tests/data/formats/SPDXJSONExample-v2.3.spdx.json`
 
 2. **CONVERTING** (for converting one format to another):
 
 * Use `pyspdxtools -i <input_file> -o <output_file>` where `<input_file>` is the location of the file to be converted
-  and `<output_file>` is the location of the output file. The output format is inferred automatically from the file ending.
-  If you are using a source distribution, try running : `pyspdxtools -i tests/data/formats/SPDXJSONExample-v2.3.spdx.json -o output.tag` 
+  and `<output_file>` is the location of the output file. The input and output formats are inferred automatically from the file endings.
+
+* If you are using a source distribution, try running:  
+  `pyspdxtools -i tests/data/formats/SPDXJSONExample-v2.3.spdx.json -o output.tag` 
 
 * If you want to skip the validation process, provide the `--novalidation` flag, like so:  
-  `pyspdxtools -i tests/data/formats/SPDXJSONExample-v2.3.spdx.json -o output.tag --novalidation`
+  `pyspdxtools -i tests/data/formats/SPDXJSONExample-v2.3.spdx.json -o output.tag --novalidation`  
+  (use this with caution: note that undetected invalid documents may lead to unexpected behavior of the tool)
+  
 * For help use `pyspdxtools --help`
 
-# Installation
-
-As always you should work in a virtualenv (venv). You can install a local clone
-of this repo with `yourenv/bin/pip install .` or install it from PyPI with
-`yourenv/bin/pip install spdx-tools`. Note that on Windows it would be `Scripts`
-instead of `bin`.
+## Library usage:
+1. **DATA MODEL**
+  * The `src.spdx.model` package constitutes the internal SPDX v2.3 data model (v2.2 is a simply a subset of this).
+  * SPDX objects are implemented via `@dataclass_with_properties`, a custom extension of `@dataclass`.
+    * Each class starts with a list of its properties and their possible types. When no default value is provided, the property is mandatory and must be set during initialization.
+    * Type checking is enforced from the type hints when initializing a new instance or setting/getting a property on an instance
```

Comment:
```suggestion
    * Type checking is enforced by the type hints when initializing a new instance or setting/getting a property on an instance
```

Reply:
I will go with "Using the type hints, type checking is enforced when ..."